unit_tests/test_users.py

```diff
@@ -33,6 +33,10 @@ def test_configure_admin_custom_password(self):
         If a password is provided, it's used to configure the admin user.
         """
         self.fakes.juju.config["password"] = "x"
+        # TODO figure out why self.fakes.juju setting is not reliably working.
```

Comment:
Can you clarify this comment? I'll try to given an explanation or maybe fix an bug in the underlying machinery.